pkg/controller/tidbcluster/pod_control.go

```diff
@@ -484,21 +486,6 @@ func needPDLeaderTransfer(pod *corev1.Pod) (string, bool) {
 	return value, true
 }
 
-func safeToRestartPD(tc *v1alpha1.TidbCluster) bool {
-	healthCount := 0
-	for _, pdMember := range tc.Status.PD.Members {
-		if pdMember.Health {
-			healthCount++
-		}
-	}
-	for _, pdMember := range tc.Status.PD.PeerMembers {
-		if pdMember.Health {
-			healthCount++
-		}
-	}
-	return healthCount > (len(tc.Status.PD.Members)+len(tc.Status.PD.PeerMembers))/2+1
```

Comment:
If we have 5 PD members and 1 of them is not healthy, then PD can't be restarted before, but now, it will be blocked when restarting, right?

Reply:
I mean,  safeToRestartPD` is replaced with the new `pdapi.IsPDStable`, but `pdapi.IsPDStable` will block the restarting process if any member is unhealthy, then the previous behavior is changed.

Reply:
Can we keep the previous behavior is `tidb.pingcap.com/pd-check-all-peers-up-before-upgrade` is not set?

Reply:
Second to @csuzhangxc, maybe keep the original `safeToRestartPD`, and when `tidb.pingcap.com/pd-check-quorum-before-upgrade` is set, keep using CR status for quorum check, and use pd client instead when the annotation is set? Less behavior changes means less surprises to us :D